projectmanagement/src/main/java/life/qbic/projectmanagement/domain/project/experiment/ExperimentalVariable.java

```diff
@@ -85,9 +92,10 @@ Result<VariableLevel, Exception> addLevel(ExperimentalValue experimentalValue) {
   }
 
   private boolean hasDifferentUnitAsExistingLevels(ExperimentalValue experimentalValue) {
-    Predicate<ExperimentalValue> hasDifferentUnit = input -> !input.unit()
-        .equals(experimentalValue.unit());
-    return levels.stream().anyMatch(hasDifferentUnit);
+    if(levels.isEmpty())
```

Comment:
dangerous. if there is already a level with a unit you would give your ok here although the experimental value has a different unit (being no unit)

Reply:
levels.isEmpty() does not check the unit, it checks if there **are** already levels at all. the first level that is added is always okay.